### PR TITLE
Initial check-in of dialog generation.

### PR DESCRIPTION
**Pull request by chrimc62**
This defines dialog:generate and is marked as preview.

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
   "dependencies": {
     "@microsoft/bf-chatdown": "1.0.0",
     "@microsoft/bf-cli-config": "1.0.0",
+    "@microsoft/bf-dialog": "1.0.0",
```
Comment (munozemilio) [inline packages/cli/package.json]: Is this going to be part of the 4.7 release?

File: packages/dialog/src/commands/dialog/generate.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                (type, msg) => {
+                    if (type === gen.FeedbackType.message
+                        || type === gen.FeedbackType.error
+                        || (type === gen.FeedbackType.info && flags.verbose)) {
+                        this.progress(msg)
+                    }
+                })
+            return true;
+        } catch (e) {
+            this.thrownError(e)
```
Comment (munozemilio) [inline packages/dialog/src/commands/dialog/generate.ts]: You can throw CLIError imported from @microsoft/bf-cli-command and have the same behavior

File: packages/dialog/src/commands/dialog/generate.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            let outDir = flags.output
+            if (!outDir) {
+                outDir = ppath.join(formName + '-resources')
+            }
+            await gen.generate(args.schema, outDir, flags.schema, flags.locale, flags.templates, flags.force,
+                (type, msg) => {
+                    if (type === gen.FeedbackType.message
+                        || type === gen.FeedbackType.error
+                        || (type === gen.FeedbackType.info && flags.verbose)) {
+                        this.progress(msg)
```
Comment (munozemilio) [inline packages/dialog/src/commands/dialog/generate.ts]: Can we use this.error directly?

File: packages/cli/package.json (hunk 1 above)
Comment (chrimc62) [reply packages/cli/package.json]: As I understand the plan is to mark the dialog stuff as [Preview], but release as part of 4.7.

File: packages/dialog/src/commands/dialog/generate.ts (hunk 3 above)
Comment (chrimc62) [reply packages/dialog/src/commands/dialog/generate.ts]: No.  The errors come from lots of different sources outside this object.

**Pull request merged**